src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -271,16 +271,7 @@ def _normalize_properties(self, user, subscriptions, is_service_principal, cert_
             if subscription_dict[_SUBSCRIPTION_NAME] != _TENANT_LEVEL_ACCOUNT_NAME:
                 if hasattr(s, 'home_tenant_id'):
                     subscription_dict[_HOME_TENANT_ID] = s.home_tenant_id
-                if hasattr(s, 'managed_by_tenants'):
-                    if s.managed_by_tenants is None:
-                        # managedByTenants is missing from the response. This is a known service issue:
-                        # https://github.com/Azure/azure-rest-api-specs/issues/9567
-                        # pylint: disable=line-too-long
-                        raise CLIError("Invalid profile is used for cloud '{cloud_name}'. "
-                                       "To configure the cloud profile, run `az cloud set --name {cloud_name} --profile <profile>(e.g. 2019-03-01-hybrid)`. "
-                                       "For more information about using Azure CLI with Azure Stack, see "
-                                       "https://docs.microsoft.com/azure-stack/user/azure-stack-version-profiles-azurecli2"
-                                       .format(cloud_name=self.cli_ctx.cloud.name))
+                if getattr(s, 'managed_by_tenants', None) is not None:
```

Comment:
- If `managed_by_tenants` doesn't exist, `getattr` returns `None` and this `if` evaluates to `False`
- If `managed_by_tenants` is `[]`, `getattr` returns `[]` and this `if` evaluates to `True`, so `[]` is preserved

Reply:
2 years later, I feel it was lucky that we didn't merge this PR.

There are 3 conditions:

```py
# Condition 1: managedByTenants missing
{

}

# Condition 2: managedByTenants is null
{
    "managedByTenants": null
}

# Condition 3: managedByTenants is empty list []
{
    "managedByTenants": []
}
```

This statement doesn't take condition 2 `"managedByTenants": null` into consideration. If ARM returns `"managedByTenants": null`, Azure CLI output will omit `managedByTenants`.

Also, for condition 1, the absence of `managedByTenants` from ARM's [Subscriptions - List](https://learn.microsoft.com/en-us/rest/api/resources/subscriptions/list) REST API will propagate all the way to Azure CLI output, causing breaking change.